core/spec.md

```diff
@@ -752,8 +845,24 @@ Support for any particular API defined by this specification is OPTIONAL,
 however it is STRONGLY RECOMMENDED that server-side implementations support at
 least the "read" (e.g. HTTP `GET`) operations. Implementations MAY choose to
 incorporate authentication and/or authorization mechanisms for the APIs.
-If an API is not supported by the server then a `405 Method Not Allowed`
-HTTP response MUST be generated.
+
+If an OPTIONAL HTTP path is not supported by an implementation, then any
+use of that API MUST generate a `404 Not Found` error.
+
+If a HTTP method is not supported for a defined HTTP path then a
+`405 Method Not Allowed` error MUST be generated.
+
+Implementation MAY defined extension APIs however the following rules MUST
+apply:
+- New HTTP paths that extend non-root paths MUST NOT be defined.
+- New root HTTP paths MAY be defined as long as they do not use Registry level
+  HTTP paths or attribute names. This include extension and GROUPS collection
+  attribute names.
+- Additional HTTP methods for specification defined HTTP paths MUST NOT be
+  defined.
+
+For example, a new API with an HTTP path of `/my-api` is allowed, but APIs with
```

Comment:
Doesn't this mean that the specification can never add any new root paths without that being a breaking change?
I wonder whether it's worth reserving a URI space - maybe even `/extensions/` and saying everything has to be under that?

Reply:
Why would adding a new root path to the spec be a breaking change, as long as it's optional ?

Or instead of "breaking" did you mean "potentially conflicting with an extension" ? Yes that can happen, but that's true of all extensions (even attribute extensions). It's just the nature of how extensions work... unless we create "extension buckets/spaces" as you suggest. But then we're back to this (very long winded) "X-" prefix on http header extensions rathole ;-)

Reply:
> Or instead of "breaking" did you mean "potentially conflicting with an extension" ?

Yes, as I view that as a breaking change - it means that doing something that *was* valid is no longer valid. Surely that's reasonable to count as "breaking".

And yes, this is like the CloudEvents discussion we had - basically I personally thing CE went the wrong route here, as I think we've now come to the conclusion we can't add *any* core attributes without taking a 2.0. I'd like xregistry to avoid getting into that situation. (I seem to remember writing a lot about this before - https://docs.google.com/document/d/1n91IEyB8gD8JSBiAYyYo8YbDne4KD3YvCa5BrdemZt4/edit?tab=t.0#heading=h.yflgqnc5ygl0 may be what I'm thinking of, or there could be something else.)